### PR TITLE
Add HBSessionStorage

### PR DESCRIPTION
**Pull request by adam-fowler**
- This moves the session storage out of SessionManager which requires a `HBRequest` to be constructed into a new type `HBSessionStorage`.
- Also added new function `getSession` as a requirement for `HBSessionAuthenticator` with a default that returns `HBRequest.session.load()`.
- Added test `testSessionStorageOutsideApp`

### CODE REVIEW

**Comment by adam-fowler** [comment]
Currently needs a new version of hummingbird before this will compile

File: Sources/HummingbirdAuth/Sessions/Request+session.swift [hunk 1]
```diff
@@ -12,13 +12,11 @@
 //
 //===----------------------------------------------------------------------===//
 
-import ExtrasBase64
-import Foundation
 import Hummingbird
 import HummingbirdFoundation
 
 /// Manage session ids and associated data
-public struct SessionManager: Sendable {
```
Comment (adam-fowler) [inline Sources/HummingbirdAuth/Sessions/Request+session.swift:21]: This should never have been set to conform to `Sendable`. Given it was only ever accessed via `HBRequest.session` I think it is safe to remove this conformance.

File: Sources/HummingbirdAuth/AsyncAwaitSupport/UnsafeTransfer.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @usableFromInline
+    var wrappedValue: Wrapped
+
+    @inlinable
+    init(_ wrappedValue: Wrapped) {
+        self.wrappedValue = wrappedValue
+    }
+}
+
+#if swift(>=5.5) && canImport(_Concurrency)
```
Comment (Joannis) [inline Sources/HummingbirdAuth/AsyncAwaitSupport/UnsafeTransfer.swift:64]: `HBUnsafeMutableTransferBox` should also support conditional conformances for `Equatable` and `Hashable`

File: Sources/HummingbirdAuth/Sessions/Application+session.swift [hunk 3]
```diff
@@ -29,7 +29,7 @@ extension HBApplication {
     ///   - sessionID: Where session id is stored in request/response
     public func addSessions(
         using storage: HBPersistDriverFactory,
-        sessionID: SessionManager.SessionIDStorage = .cookie("HB_SESSION_ID")
+        sessionID: HBSessionStorage.SessionIDStorage = .cookie("HB_SESSION_ID")
```
Comment (Joannis) [inline Sources/HummingbirdAuth/Sessions/Application+session.swift:32]: This is a breaking change right?

File: Sources/HummingbirdAuth/Sessions/Application+session.swift [hunk 4]
```diff
@@ -40,7 +40,7 @@ extension HBApplication {
     /// Add session management to `HBApplication` using default persist memory driver
     /// - Parameter sessionID: Where session id is stored in request/response
     public func addSessions(
-        sessionID: SessionManager.SessionIDStorage = .cookie("HB_SESSION_ID")
+        sessionID: HBSessionStorage.SessionIDStorage = .cookie("HB_SESSION_ID")
```
Comment (Joannis) [inline Sources/HummingbirdAuth/Sessions/Application+session.swift:43]: This is a breaking change right?

File: Sources/HummingbirdAuth/Sessions/Application+session.swift (hunk 3 above)
Comment (adam-fowler) [reply Sources/HummingbirdAuth/Sessions/Application+session.swift:32]: yeah fair point

File: Sources/HummingbirdAuth/AsyncAwaitSupport/UnsafeTransfer.swift (hunk 2 above)
Comment (adam-fowler) [reply Sources/HummingbirdAuth/AsyncAwaitSupport/UnsafeTransfer.swift:64]: Its already `Equatable` and `Hashable` as `HBUnsafeMutableTransferBox` is a class

File: Sources/HummingbirdAuth/Sessions/Application+session.swift (hunk 3 above)
Comment (adam-fowler) [reply Sources/HummingbirdAuth/Sessions/Application+session.swift]: Breaking change has been fixed

**Pull request merged**